src/modules/connection.js

```diff
@@ -16,39 +16,38 @@ class Connection {
    * Connect to the broker. We keep only 1 connection for each connection string provided in config, as advised by RabbitMQ
    * @return {Promise} A promise that resolve with an amqp.node connection object
    */
-  getConnection() {
-    // cache handling, if connection already opened, return it
-    if (this._connection) {
-      return this._connection;
-    }
+  async getConnection() {
+    try {
+      // cache handling, if connection already opened, return it
+      if (this._connection) {
+        return this._connection;
+      }
 
-    // prepare the connection internal object, and reset channel if connection has been closed
-    this._connection = amqp
-      .connect(this._config.host, {
+      // prepare the connection internal object, and reset channel if connection has been closed
+      const connection = await amqp.connect(this._config.host, {
```

Comment:
Here and in channels, I think you change it's behaviour. The idea of the existing code was to cache the promise it self, so if multiple calls to "getConnection" (or channel) will happen they will all wait on the same promise.